connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -357,6 +367,62 @@ public void validateConnectorConfig(Map<String, String> connectorProps, Callback
         });
     }
 
+    /**
+     * Build the {@link RestartPlan} that describes what should and should not be restarted given the restart request
+     * and the current status of the connector and task instances.
+     *
+     * @param request the restart request; may not be null
+     * @return the restart plan, or empty if this worker has no status for the connector named in the request and therefore the
+     *         connector cannot be restarted
+     */
+    public Optional<RestartPlan> buildRestartPlan(RestartRequest request) {
+        String connectorName = request.connectorName();
+        ConnectorStatus connectorStatus = statusBackingStore.get(connectorName);
+        if (connectorStatus == null) {
+            return Optional.empty();
+        }
+
+        // If requested, mark the connector as restarting
+        AbstractStatus.State connectorState;
+        if (request.shouldRestartConnector(connectorStatus)) {
+            connectorState = AbstractStatus.State.RESTARTING;
+        } else {
+            connectorState = connectorStatus.state();
+        }
```

Comment:
nit: the ternary operator can be used (`?:`) as below, unless you're not a fan. 
```suggestion
        AbstractStatus.State connectorState = request.shouldRestartConnector(connectorStatus)
                ? AbstractStatus.State.RESTARTING 
                : connectorStatus.state();
```

Reply:
fixed